### PR TITLE
Whipped cream and frosting requirements

### PR DESCRIPTION
**Pull request by Maleclypse**
#### Summary
Content "Adds Whipped cream and any_frosting requirement"

#### Purpose of change
Add's additional use for heavy cream and makes it interchangeable with buttercream.

#### Describe the solution
Add's whipped cream, recipe for it and requirements to use it.  Corrects cake and homemade toaster pastry to use new any_frosting requirement.

#### Describe alternatives you've considered
Are there any other recipes that should be adjusted to include frosting?  Are there any additional recipes that should be created to use frosting?

#### Testing
TBD

### CODE REVIEW

**Comment by Maleclypse** [comment]
I apologize that this has autopulled in so many code reviews.  I think it's from my repo's commit history.

File: data/json/recipes/recipe_food.json [hunk 1]
```diff
@@ -5433,7 +5433,7 @@
     "charges": 1,
     "autolearn": true,
     "tools": [ [ [ "surface_heat", 10, "LIST" ] ] ],
-    "components": [ [ [ "toastem4", 1 ] ], [ [ "buttercream", 1 ] ] ]
+    "components": [ [ [ "toastem4", 1 ] ], [ [ "any_frosting", 1 ] ] ]
```
Comment (actual-nh) [inline data/json/recipes/recipe_food.json:5436]:
```suggestion
    "components": [ [ [ "toastem4", 1 ] ], [ [ "any_frosting", 1, "LIST" ] ] ]
```

File: data/json/items/comestibles/dairy.json [hunk 2]
```diff
@@ -296,6 +296,28 @@
...
+    "id": "milk_cream_whipped",
+    "name": { "str_sp": "whipped cream" },
+    "weight": "120 g",
+    "color": "white",
+    "spoils_in": "1 day",
+    "comestible_type": "FOOD",
+    "symbol": "~",
+    "quench": 3,
+    "calories": 250,
+    "description": " Cream that has been whipped, perfect for frosting cakes and pies.  Either store bought or homemade.",
```
Comment (casswedson) [inline data/json/items/comestibles/dairy.json:310]:
```suggestion
    "description": "Cream that has been whipped, perfect for frosting cakes and pies.  Either store bought or homemade.",
```

File: data/json/items/comestibles/dairy.json [hunk 3]
```diff
@@ -297,6 +297,28 @@
...
+    "id": "milk_cream_whipped",
+    "name": { "str_sp": "whipped cream" },
+    "weight": "120 g",
+    "color": "white",
+    "spoils_in": "1 day",
+    "comestible_type": "FOOD",
+    "symbol": "~",
+    "quench": 3,
+    "calories": 250,
+    "description": "Cream that has been whipped, perfect for frosting cakes and pies.  Either store bought or homemade.",
```
Comment (actual-nh) [inline data/json/items/comestibles/dairy.json:311]:
```suggestion
    "description": "Cream that has been whipped, perfect for frosting cakes and pies.  Either store-bought or homemade.",
```

File: data/json/items/ranged/slings.json [hunk 4]
```diff
@@ -79,8 +79,8 @@
     "ammo": [ "rock" ],
     "weight": "2000 g",
     "volume": "3 L",
-    "longest_side": "180 cm",
     "price_postapoc": 250,
+    "longest_side": "180 cm",
```
Comment (actual-nh) [inline data/json/items/ranged/slings.json:83]: ?

**Review by actual-nh — COMMENTED** [review]
Nice! Should perhaps give ideas for My Sweet Cataclysm (@Fris0uman? Still need to go over the faction relationships for it, BTW.)

File: data/json/items/ranged/slings.json (hunk 4 above)
Comment (Maleclypse) [inline data/json/items/ranged/slings.json:83]:
```suggestion

```

File: data/json/items/ranged/slings.json [hunk 5]
```diff
@@ -79,8 +79,8 @@
     "ammo": [ "rock" ],
     "weight": "2000 g",
     "volume": "3 L",
-    "longest_side": "180 cm",
```
Comment (Maleclypse) [inline data/json/items/ranged/slings.json:82]:
```suggestion
    "longest_side": "180 cm",
```

**Pull request merged**